### PR TITLE
Correct BCI for NewArray and NewMultiArray Statements and Fix DaCapo

### PR DESCRIPTION
**Pull request by dheeraj135**
Major Changes:
1. Use Correct BCI for New Array and New MultiArray Statements.
2. Change Output Format
3. Print only Internal object types.
4. Fix Dacapo issues ( don't analyse JDK and fix Null/ OutofBounds errors. )

### CODE REVIEW

File: src/utils/getBCI.java [hunk 1]
```diff
@@ -1,12 +1,30 @@
...
+		for(ValueBox ub: u.getUseBoxes() ) {
+			if(!ub.getClass().toString().equals("class soot.jimple.internal.JAssignStmt$LinkedRValueBox"))
+                continue;
+            Value v = ub.getValue();
+            if(v==null)
+                continue;
+            if(v instanceof JNewArrayExpr || v instanceof JNewMultiArrayExpr )
+                ;
+			else continue;
+			BytecodeOffsetTag tg = (BytecodeOffsetTag) ub.getTag("BytecodeOffsetTag");
```
Comment (42niks) [inline src/utils/getBCI.java:51]: code structure/formatting is making it hard to read. Could you please add another commit making it smoother to read? shouldn't take long.
Also please restructure
`if(condition) <nothing> else continue`
to
`if(not condition) continue`

File: src/es/ConditionalValue.java [hunk 2]
```diff
@@ -102,13 +102,14 @@ public boolean compareAtDepth(ConditionalValue e, int depth) {
...
 				if (this.fieldList != null && this.fieldList.size() != depth) return false;
-				for (int i = 0; i < depth; i++) {
+				for (int i = 0; i < depth && i< e.fieldList.size() ; i++) {
 					if (!this.fieldList.get(i).equals(e.fieldList.get(i))) return false;
 				}
 			}
 		} catch (Exception e2) {
 			System.out.println("Comparing:" + this.toString() + " with " + e.toString() + " at depth " + depth);
-			throw new IllegalArgumentException();
+			// throw new IllegalArgumentException();
```
Comment (42niks) [inline src/es/ConditionalValue.java]: delete this line please

**Review by 42niks — APPROVED** [review]

File: src/es/ConditionalValue.java (hunk 2 above)
Comment (dheeraj135) [reply src/es/ConditionalValue.java]: Done.

File: src/utils/getBCI.java (hunk 1 above)
Comment (dheeraj135) [reply src/utils/getBCI.java:51]: Done.

**Comment by dheeraj135** [comment]
Further changes:
1. Use com.apache.* instead of com.apache. Also, excluded the jdk.* classes.
2. We should use ConcurrentHashMap instead of standard thread-unsafe HashMap because multiple threads change the HashMap at the same time.

**Comment by dheeraj135** [comment]
I am merging this PR. If we find some issue, we can always fix it later.

**Pull request merged**